deriva/core/ermrest_model.py

```diff
@@ -2030,11 +2044,19 @@ def alter(
 
         return self
 
-    def drop(self):
+    def drop(self, cascade=False):
         """Remove this column from the remote database.
+
+        :param cascade: drop dependent objects.
         """
         if self.name not in self.table.column_definitions.elements:
             raise ValueError('Column %s does not appear to belong to table %s.' % (self, self.table))
+
+        if cascade:
+            for key in list(self.table.keys):
```

Comment:
We need the same loop over self.foreign_keys to check whether self is in fkey.foreign_key_columns, don't we?

Reply:
Yes. I missed that!

Reply:
added fkey loop